test/array.test.js

```diff
@@ -300,10 +309,9 @@ test('Array flat', () => {
 
 test('Array flatMap', () => {
   let a = newArray();
-  a.set(3, newArray().contents);
```

Comment:
Please leave this line in. The whole purpose of the test is to test nested arrays.

Reply:
That's not how `flatMap` works; it flattens arrays in the result of `map` (to depth 1).

Reply:
No matter what the expected result is, we should test an array as array item here.

Reply:
I've added the array back in, but I've changed the callback function (again) because it doesn't make sense for it never to return an array, as otherwise there would be nothing to flatten.